### PR TITLE
qa/tests: initial check in for octopus-x/parallel tests

### PR DESCRIPTION
**Pull request by yuriw**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yuriw** [comment]
ref: https://github.com/ceph/ceph/pull/35808

**Comment by yuriw** [comment]
`CEPH_BRANCH=master; MACHINE_NAME=smithi; teuthology-suite -v -c $CEPH_BRANCH --ceph-repo https://github.com/ceph/ceph.git --suite-repo https://github.com/yuriw/ceph.git -k distro -n 7 -m $MACHINE_NAME -s upgrade/octopus-x -t master -e $CEPH_QA_EMAIL -p 50 --suite-branch wip-yuriw-octopus-x-parallel-master -l 2`

http://pulpito.ceph.com/teuthology-2020-08-12_23:49:10-upgrade:octopus-x-master-distro-basic-smithi/

**Comment by yuriw** [comment]
http://pulpito.ceph.com/teuthology-2020-10-30_15:49:07-upgrade:octopus-x-master-distro-basic-smithi/

@neha-ojha FYI

**Comment by yuriw** [comment]
@jdurgin @neha-ojha @sebastian-philipp this PR seems stuck

@sebastian-philipp any suggestions/fixes we can try ?

**Comment by yuriw** [comment]
trying again 
http://pulpito.ceph.com/teuthology-2020-12-04_19:57:53-upgrade:octopus-x-master-distro-basic-smithi/

**Comment by yuriw** [comment]
https://tracker.ceph.com/issues/48469

**Comment by yuriw** [comment]
testing on @dmick 's wip-48469

`CEPH_BRANCH=master; MACHINE_NAME=smithi; teuthology-suite -v -c $CEPH_BRANCH --ceph-repo https://github.com/ceph/ceph.git --suite-repo https://github.com/yuriw/ceph.git -k distro -n 7 -m $MACHINE_NAME -s upgrade/octopus-x -t wip-48469 -e $CEPH_QA_EMAIL -p 50 --suite-branch wip-yuriw-octopus-x-parallel-master -l 2`

CEPH_BRANCH=master; MACHINE_NAME=smithi; teuthology-suite -v -c $CEPH_BRANCH --ceph-repo https://github.com/ceph/ceph.git --suite-repo https://github.com/yuriw/ceph.git -k distro -n 7 -m $MACHINE_NAME -s upgrade/octopus-x -t wip-48469 -e $CEPH_QA_EMAIL -p 50 --suite-branch wip-yuriw-octopus-x-parallel-master -l 2

**Comment by yuriw** [comment]
rerunning with `roleless: false`

http://pulpito.ceph.com/teuthology-2020-12-16_00:42:09-upgrade:octopus-x-master-distro-basic-smithi/

File: qa/suites/upgrade/octopus-x/parallel/0-cluster/start.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- - mon.a #temp as roleless disabled
+  - mgr.x #temp as roleless disabled
+  - mds.a
+  - osd.0
+  - osd.1
+  - osd.2
+  - osd.3
+  - client.0
+  - client.1
+- - host.b
```
Comment (neha-ojha) [inline qa/suites/upgrade/octopus-x/parallel/0-cluster/start.yaml:17]: @yuriw please comment this out too, when not using roleless, we don't need host.x

File: qa/suites/upgrade/octopus-x/parallel/0-cluster/start.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- - mon.b #temp as roleless disabled
+  - osd.4
+  - osd.5
+  - osd.6
+  - osd.7
+  - client.2
+  - client.3
+- - host.c
```
Comment (neha-ojha) [inline qa/suites/upgrade/octopus-x/parallel/0-cluster/start.yaml:25]: ditto

**Comment by yuriw** [comment]
Fixed nodes https://hastebin.com/riqinicuyu and testing again:

http://pulpito.ceph.com/teuthology-2020-12-16_22:05:18-upgrade:octopus-x-master-distro-basic-smithi/

**Comment by yuriw** [comment]
see https://github.com/ceph/ceph/pull/38707

**Pull request closed without merging**